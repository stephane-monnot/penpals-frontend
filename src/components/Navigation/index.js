import React from 'react';

const Navigation = ({ children }) => (
  <ul className="navigation">
    {children}
  </ul>
);

export default Navigation;
