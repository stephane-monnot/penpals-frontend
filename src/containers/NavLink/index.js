import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const NavLink = (props) => (
  <li className="navigation-link">
    <Link {...props} />
  </li>
);

NavLink.propTypes = {
  to: PropTypes.string.isRequired
};

export default NavLink;
