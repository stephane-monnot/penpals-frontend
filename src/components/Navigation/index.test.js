import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import Navigation from './';
import NavLink from '../NavLink';

describe(Navigation, () => {
  const component = mount(
    <MemoryRouter location="someLocation" context={{}}>
      <Navigation>
        <NavLink to="/home" label="Home" />
        <NavLink to="/about" label="About" />
      </Navigation>
    </MemoryRouter>
  )

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <MemoryRouter location="someLocation" context={{}}>
        <Navigation>
          <NavLink to="/home" label="Home" />
          <NavLink to="/about" label="About" />
        </Navigation>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains a `ul` subcomponent with a class `navigation`', () => {
    expect(component.find('ul')).toHaveLength(1);
    expect(component.find('ul').hasClass('navigation')).toEqual(true);
  });

  it('renders children', () => {
    expect(component.find('li')).toHaveLength(2);
  });
});
