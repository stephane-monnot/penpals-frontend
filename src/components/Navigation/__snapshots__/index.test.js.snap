// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`({ children }) =>
_react2.default.createElement("ul", { className: "navigation", __source: { fileName: _jsxFileName, lineNumber: 4 }, __self: undefined },
  children) renders and matches our snapshot 1`] = `
<ul
  className="navigation"
>
  <li
    className="navigation-link"
  >
    <a
      href="/home"
      label="Home"
      onClick={[Function]}
    />
  </li>
  <li
    className="navigation-link"
  >
    <a
      href="/about"
      label="About"
      onClick={[Function]}
    />
  </li>
</ul>
`;
